File: demos/python-feequote/feequote/legacy/fees.py
"""Retained for the nightly batch reconciliation job. Scheduled for removal once
reconciliation moves to core pricing (see ticket 4821).
"""

from __future__ import annotations


def calculate_fee(amount_cents: int, rate_bps: int) -> int:
    """Fee at ``rate_bps`` for ``amount_cents``, truncated to whole cents."""
    return amount_cents * rate_bps // 10_000

File: demos/java-feequote/src/main/java/com/feequote/legacy/Fees.java
package com.feequote.legacy;

/**
 * Retained for the nightly batch reconciliation job. Scheduled for removal once
 * reconciliation moves to core pricing (see ticket 4821).
 */
public final class Fees {

    private Fees() {
    }

    /** Fee at {@code rateBps} for {@code amountCents}, truncated to whole cents. */
    public static long calculateFee(long amountCents, int rateBps) {
        return amountCents * rateBps / 10_000;
    }
}
